code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java

```diff
@@ -85,21 +86,23 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
     }
 
     protected void handleVolumeEvent(final ExternalEvent event, ProcessState state, ProcessInstance process) {
-        lockManager.lock(new ExternalEventLock(VOLUME_POOL_LOCK_NAME, event.getAccountId(), event.getExternalId()), new LockCallbackNoReturn() {
+        //new DockerStoragePoolVolumeCreateLock(storagePool, dVol.getUri())
+        Object driver = CollectionUtils.getNestedValue(DataUtils.getFields(event), FIELD_VOLUME, VolumeConstants.FIELD_VOLUME_DRIVER);
+        final Object name = CollectionUtils.getNestedValue(DataUtils.getFields(event), FIELD_VOLUME, FIELD_NAME);
+        if (driver == null || name == null) {
+            log.warn("Driver or volume name not specified. Returning. Event: {}", event);
+            return;
+        }
+        String driverName = driver.toString();
+        final StoragePool storagePool = storagePoolDao.findStoragePoolByDriverName(event.getAccountId(), driverName);
+        if (storagePool == null) {
+            log.warn("Unknown storage pool. Returning. Driver name: {}", driverName);
+            return;
+        }
+
+        lockManager.lock(new DockerStoragePoolVolumeCreateLock(storagePool, event.getExternalId()), new LockCallbackNoReturn() {
```

Comment:
The important thing here is that I changed the lock from `ExternalEventLock` to `DockerStoragePoolVolumeCreateLock`. The other code moved so that I could get the storagePool before creating the lock, which is a parameter to the lock.